### PR TITLE
Add word2vec code and python notebook to test sentiment analysis. Req…

### PR DESCRIPTION
**Pull request by mlomnitz**
…uires tensorflow to run

### CODE REVIEW

**Review by sam-dixon — APPROVED** [review]

File: Notebooks/word2vec/Load_Text_Set.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+# Read data from a downloaded .zip
+def read_data(file):
+    with zipfile.ZipFile(file) as f:
+        data = tf.compat.as_str(f.read(f.namelist()[0])).split()
+    return data
+
+# Data object
+class text_8(object):
+    def __init__(self,  vocabulary = 200000):
```
Comment (cindeem) [inline Notebooks/word2vec/Load_Text_Set.py:25]: Small feedback comments:
- __init__ functions should be lightweight....and not do much processing....
instantiating a class should only involve very lightweight processes

File: Notebooks/word2vec/Load_Text_Set.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        filename  = 'text8.zip'
+        # Init the data set, if file is absent download it
+        if not os.path.exists(filename):
+            print('File is missing, retrieving ', url+filename)
+            filename, _ = urlretrieve(url+filename,filename)
+        # Verify that datasert was correctly loaded by checking bits.
+        if os.stat(filename).st_size != expected_bits:
+            print('Cannot load & verify. Please sort this out')
```
Comment (cindeem) [inline Notebooks/word2vec/Load_Text_Set.py:33]: This should raise an exception
https://docs.python.org/3/tutorial/errors.html

File: Notebooks/word2vec/Load_Text_Set.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            return
+        print('Found and verified %s ' % filename)
+        words = read_data(filename)
+        print('Words in file ', len(words))
+        self.data, self.count, self.dictionary, self.reverse_dictionary = self.build_dataset(words, vocabulary)
+    # Build dictionary of most common words from the loaded word set. This is done to speed up and reduce memory usage.
+    def build_dataset(self, words, vocabulary):
```
Comment (cindeem) [inline Notebooks/word2vec/Load_Text_Set.py:40]: I know Python is not your language, here is a useful set of guidelines
https://www.python.org/dev/peps/pep-0008/
doc strings on functions are nice

File: Notebooks/word2vec/tf_Word2Vec.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+from itertools import compress
+
+# Some model constants
+data_index = 0
+embedding_size = 128 # Dimension of the embedding vector.
+num_sampled = 64 # Number of negative examples to sample.
+
+# Generate the batch and associated labels for training in skipgram
+def generate_batch(batch_size, num_skips, skip_window, data):
+    global data_index
```
Comment (cindeem) [inline Notebooks/word2vec/tf_Word2Vec.py:26]: globals are considered a bit heavy handed in python, might be good to explain why you explicitly make this a global

**Pull request merged**